src/Interpreters/SystemLog.h

```diff
@@ -272,7 +276,7 @@ void SystemLog<LogElement>::add(const LogElement & element)
         return;
     }
 
-    queue.push_back(element);
+    queue.push_back(std::forward<T>(element));
```

Comment:
This should avoid copy-construction where possible, but not always -- depends on what kind of reference you pass. Maybe we should require rvalue reference input instead?

Also there is still reallocation of `queue` buffer itself -- we can probably grow it to max size right away. If you do this, don't forget also initialize the local queue in the flushing thread, that is swapped with this one.

Reply:
1) I thought the problem was only with `text_log`. Other logs will pass lvalue as before. 
2) That is the problem.